broker/src/main/java/io/camunda/zeebe/broker/system/partitions/ZeebePartition.java

```diff
@@ -58,8 +57,8 @@ public ZeebePartition(final PartitionContext context, final PartitionTransition
 
     actorName = buildActorName(context.getNodeId(), "ZeebePartition", context.getPartitionId());
     context.setComponentHealthMonitor(new CriticalComponentsHealthMonitor(actor, LOG));
-    raftPartitionHealth =
-        new RaftPartitionHealth(context.getRaftPartition(), actor, this::onRaftFailed);
+    raftPartitionHealth = new RaftPartitionHealth(context.getRaftPartition(), actor);
+    raftPartitionHealth.addFailureListener(this);
```

Comment:
I think we don't need this. Because later we are adding `raftPartitionHealth` to the health monitor, so it would mark the partition as unhealthy if any of the component is unhealthy.

Reply:
True, I only added it to fully use the `HealthMonitorable` (both `getHealthStatus` and the FailureListener) but I agree. I'll remove it